File: 백준/Gold/13164. 행복 유치원/행복 유치원.js
const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");        // 여러 줄 입력

let arr = input[0].split(" ");              // 사람의 수, 만들고자하는 조의 수
let person = input[1].split(" ");           // 사람의 키

let temp = [];                              // 키 차이 저장 배열
let before = Number(person[0]);             // 이전 값
for(let i = 1; i < Number(arr[0]); i++) {
    let pre = Number(person[i]);            // 현재 값
    temp[i-1] = pre - before;               // 현재 값에서 이전 값을 빼주고 그 값을 저장
    before = pre;                           // 현재 값을 이전 값에 저장
}

temp.sort((a,b) => a-b);                    // 내림차순 정렬

let sum = 0;                                // 결과값
for(let i = 0; i < Number(arr[0]) - Number(arr[1]); i++) {          // 사람의 수 - 만들고자하는 조의 수 만큼 묶음의 값을 저장
    sum += temp[i];
}

console.log(sum);                           

File: 백준/Gold/13164. 행복 유치원/행복 유치원.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer stk = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(stk.nextToken());          // 사람의 수
        int k = Integer.parseInt(stk.nextToken());          // 조의 수
        long[] temp = new long[n-1];                        // 차이를 저장할 배열
        stk = new StringTokenizer(br.readLine());
        long before = Long.parseLong(stk.nextToken());      // 이전 값
        for(int i = 0; i < n-1; i++) {
            long pre = Long.parseLong(stk.nextToken());     // 현재 값
            temp[i] = pre - before;                         // 현재 값 - 이전 값을 저장
            before = pre;                                   // 현재 값을 이전 값에 저장
        }

        Arrays.sort(temp);                                  // 차이를 저장한 배열을 정렬

        long sum = 0;                                       // 결과값
        for(int i = 0; i < n-k; i++) {                      // 사람의 수에서 만들려는 조의 수를 빼면 묶음의 수가 나옴
            sum += temp[i];                                 // 묶음의 수만큼 값을 저장
        }

        System.out.println(sum);                            // 결과값 출력
        br.close();
    }
}